packages/connector-puppeteer/src/lib/actions.ts

```diff
@@ -0,0 +1,57 @@
+import { resolve } from 'path';
+
+import * as puppeteer from 'puppeteer-core';
+import { ConnectorOptions } from '../connector';
+
+export type ActionConfig = {
+    file: string;
+    on: 'beforeTargetNavigation' | 'afterTargetNavigation';
+    options?: object;
+}
+
+type Action = (page: puppeteer.Page, config: ConnectorOptions) => Promise<void>;
+
+export type UserActions = {
+    afterTargetNavigation: Action[];
+    beforeTargetNavigation: Action[];
+};
+
+/**
+ * Loads and groups user provided actions by the moment they need
+ * to be executed (`beforeTargetNavigation` or `afterTargetNavigation`).
+ * @param actions The user configuration with the actions
+ */
+export const group = (actions: ActionConfig[] = []): UserActions => {
+    const beforeTargetNavigation: Action[] = [];
+    const afterTargetNavigation: Action[] = [];
+
+    for (const actionConfig of actions) {
+        let loadedAction: { action: Action};
+        const pathToUserAction = resolve(process.cwd(), actionConfig.file);
+
+        try {
+            loadedAction = require(pathToUserAction) as any;
```

Comment:
I don't think `as any` should be necessary here as `require` already returns an `any` type an `loadedAction` is already explicitly typed.
```suggestion
            loadedAction = require(pathToUserAction);
```

Reply:
Done